dev/build_arrow.sh

```diff
@@ -35,9 +35,15 @@ function prepare_arrow_build() {
 }
 
 function build_arrow_cpp() {
- pushd $ARROW_PREFIX/cpp
-
- cmake_install \
+  pushd $ARROW_PREFIX/cpp
+  ARROW_WITH_ZLIB=ON
+  if [[ "$(uname)" == "Darwin" ]]; then
+    clang_major_version=$(echo | clang -dM -E - | grep __clang_major__ | awk '{print $3}')
+    if [ "${clang_major_version}" -ge 17 ]; then
+      ARROW_WITH_ZLIB=OFF
```

Comment:
With ARROW_WITH_ZLIB=OFF, is there any impact on the functionalities?

Reply:
Velox can compile successfully probably because ZLIB_SOURCE=AUTO, I will try gluten later.

Reply:
This is risky. We just temporarily disable zlib and enable it after arrow is upgraded.

Reply:
We may have a better track on the mac os build after https://github.com/apache/incubator-gluten/pull/9579.